tools/windows/DatadogAgentInstaller/CustomActions/ConfigureUserCustomActions.cs

```diff
@@ -506,6 +506,78 @@ private void ConfigureFilePermissions()
+            // The above pages aren't super clear on which users can access local secrets, but the following pages
+            // show that only Administrators have POLICY_GET_PRIVATE_INFORMATION by default, and this can
+            // be confirmed empirically.
+            // https://learn.microsoft.com/en-us/windows/win32/secmgmt/policy-object-access-rights
+            // https://learn.microsoft.com/en-us/windows/win32/secmgmt/policy-object-protection
+            // https://learn.microsoft.com/en-us/windows/win32/secmgmt/private-data-object-initial-protection
+            // Machine private data objects, 'M$', are not readable (through the API) even by LocalSystem.
+            // https://learn.microsoft.com/en-us/windows/win32/secmgmt/private-data-object
+            var secretType = "L$";
+            return $"{secretType}datadog_ddagentuser_password";
+        }
+
+        private void ConfigureUserPassword()
+        {
+            var ddagentuserPassword = _session.Property("DDAGENTUSER_PROCESSED_PASSWORD");
+            var ddagentuser = _session.Property("DDAGENTUSER_PROCESSED_NAME");
+            var resetPassword = _session.Property("DDAGENTUSER_RESET_PASSWORD");
+            if (!string.IsNullOrEmpty(resetPassword))
+            {
+                _session.Log($"Resetting {ddagentuser} password.");
+                if (string.IsNullOrEmpty(ddagentuserPassword))
+                {
+                    throw new InvalidOperationException("Asked to reset password, but password was not provided");
+                }
+
+                _nativeMethods.SetUserPassword(ddagentuser, ddagentuserPassword);
+            }
+
+            // Store the password in LSA secret store so that it can be used during Fleet Automation remote upgrades
+            // This is the same place that Windows Service Manager stores service account passwords, so
+            // this isn't introducing any NEW security risks.
+            // https://docs.microsoft.com/en-us/windows/win32/services/service-accounts
+            // https://learn.microsoft.com/en-us/windows/win32/secmgmt/storing-private-data
+            var keyName = AgentPasswordPrivateDataKey();
+            var isServiceAccount = _session.Property("DDAGENTUSER_IS_SERVICE_ACCOUNT") == "true";
+            if (isServiceAccount)
+            {
+                // If ddagentuser is a service account, it has no password, so remove any previous entries from the LSA
+                _session.Log("Agent user is a service account, removing password from LSA secret store");
+                try
+                {
+                    _nativeMethods.RemoveSecret(keyName);
+                }
+                catch (Exception e)
+                {
+                    // Don't fail if we fail to remove the secret.
+                    // ProcessDDAgentUserCredentials will appropriately clear the password property for service accounts
+                    // so it being left behind shouldn't affect anything.
```

Comment:
```suggestion
                    // Don't fail if we fail to remove the secret.
                    // ProcessDDAgentUserCredentials will appropriately clear the password property for service accounts
                    // so nothing should be left behind.
```
maybe something like that

Reply:
can also leave original